app/viewmodels/SecurityStatementsViewModel.scala

```diff
@@ -16,9 +16,259 @@
+
+object SecurityStatementsViewModel {
+  def apply(statementsForAllEoris: Seq[SecurityStatementsForEori])(implicit appConfig: AppConfig,
+                                                                   messages: Messages): SecurityStatementsViewModel = {
+    SecurityStatementsViewModel(
+      pageTitle = Some(messages("cf.security-statements.title")),
+      backLink = Some(appConfig.customsFinancialsFrontendHomepage),
+      header = generateHeader,
+      requestedStatementNotification = generateRequestedStatementNotification(statementsForAllEoris),
+      currentStatements = generateCurrentStatements(statementsForAllEoris),
+      missingGuidance = generateMissingGuidance,
+      statementServiceParagraph = generateStatementServiceParagraph,
+      requestStatementsLink = generateRequestStatementLink,
+      details = SecurityStatementsViewModelDetails(
+        statementsForAllEoris = statementsForAllEoris,
+        hasRequestedStatements = hasRequestedStatements(statementsForAllEoris),
+        hasCurrentStatements = hasCurrentStatements(statementsForAllEoris)))
+  }
+
+  private def generateHeader(implicit messages: Messages): HtmlFormat.Appendable = {
+    h1Component(msg = "cf.security-statements.title", classes = "govuk-heading-xl")
+  }
+
+  private def generateRequestedStatementNotification(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                                    (implicit appConfig: AppConfig, messages: Messages
+                                                    ): HtmlFormat.Appendable = {
+    if (hasRequestedStatements(statementsForAllEoris)) {
+      new requestedStatements(linkComponent).apply(
+        url = appConfig.requestedStatements(SecurityStatement))
+    } else {
+      HtmlFormat.empty
+    }
+  }
+
+  private def generateCurrentStatements(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                       (implicit messages: Messages): HtmlFormat.Appendable = {
+    if (hasCurrentStatements(statementsForAllEoris)) {
+      HtmlFormat.fill(statementsForAllEoris.zipWithIndex.flatMap {
+        case (statementsForEori, historyIndex) if statementsForEori.currentStatements.nonEmpty =>
+          val eoriHeader = generateEoriHeader(historyIndex, statementsForEori)
+          val eomHeader = generateEomHeader(statementsForEori)
+
+          val pdfStatements = generateStatements(statementsForEori, historyIndex, isCsv = false)
+          val csvStatements = generateStatements(statementsForEori, historyIndex, isCsv = true)
+
+          Seq(eoriHeader, Some(pdfStatements), eomHeader, Some(csvStatements)).flatten
+
+        case _ => Seq.empty
+      })
+    } else {
+      generateNoStatementsParagraph
+    }
+  }
+
+  private def generateEoriHeader(historyIndex: Int, statementsForEori: SecurityStatementsForEori)
+                                (implicit messages: Messages): Option[HtmlFormat.Appendable] = {
+    if (historyIndex > 0) {
```

Comment:
Sometimes we're using if statements and sometimes case statements in these private methods, do we want to just use one if possible and be consistent?